tests/utils/test_usage_stats.py

```diff
@@ -30,3 +36,56 @@ class ExampleBentoService(bentoml.BentoService):
     assert properties["artifact_types"]
     assert 'NO_ARTIFACT' in properties["artifact_types"]
     assert properties["env"] is not None
+
+
+def mock_track_cli(event, properties):
+    print(event)
+    print(json.dumps(properties))
+
+
+@patch('bentoml.cli.click_utils.track_cli', MagicMock(side_effect=mock_track_cli))
+def test_track_cli_usage(bento_service, bento_bundle_path):
+    runner = CliRunner()
+    cli = create_bentoml_cli()
+    successful_result = runner.invoke(
+        cli.commands['info'], [f'{bento_service.name}:{bento_service.version}']
+    )
+    track_result = successful_result.output.strip().split('\n')[-2:]
+    assert successful_result.exit_code == 0
+    assert track_result[0] == 'info'
+    additional_info = json.loads(track_result[1])
+    assert additional_info['duration']
+
+
+@patch('bentoml.cli.click_utils.track_cli', MagicMock(side_effect=mock_track_cli))
+def test_track_cli_with_click_exception():
+    runner = CliRunner()
+    cli = create_bentoml_cli()
+    failed_result = runner.invoke(
+        cli.commands['azure-functions'], ['update', 'mock-deployment-name']
+    )
+    track_result = failed_result.output.strip().split('\n')
+    print(track_result)
+    assert failed_result.exit_code == 1
+    event_name = track_result[0].split('\r')[1]
+    assert event_name == 'deploy-update'
+    additional_info = json.loads(track_result[1])
+    assert additional_info['platform'] == 'AZURE_FUNCTIONS'
+
+
+def mock_bento_api_server(bento_service, port):
+    raise KeyboardInterrupt()
+
+
+@patch('bentoml.cli.click_utils.track_cli', MagicMock(side_effect=mock_track_cli))
```

Comment:
you should verify that track_cli got called by the expected parameter.

not the return value of runner.invoke

Reply:
let's not make this test depend on the output of the CLI command